detox/runners/jest-circus/listeners/DetoxInitErrorListener.js

```diff
@@ -1,19 +1,20 @@
 const _ = require('lodash');
 
 class DetoxInitErrorListener {
-  setup(event, { unhandledErrors, rootDescribeBlock }) {
-    if (unhandledErrors.length > 0) {
-      rootDescribeBlock.mode = 'skip';
+  add_hook(event, { unhandledErrors, currentDescribeBlock, rootDescribeBlock }) {
+    if (_.isEmpty(unhandledErrors)) {
+      return;
+    }
+
+    if (currentDescribeBlock !== rootDescribeBlock || currentDescribeBlock.hooks.length > 1) {
+      _.last(currentDescribeBlock.hooks).fn = async () => {};
     }
   }
 
-  add_test(event, { currentDescribeBlock, rootDescribeBlock }) {
-    if (currentDescribeBlock === rootDescribeBlock && rootDescribeBlock.mode === 'skip') {
+  add_test(event, { unhandledErrors, currentDescribeBlock }) {
```

Comment:
Just making sure: Do we know for a fact that `unhandledErrors` is synonymous with errors happening prior to a describe-suite execution? Could they, for example, be a result of failures of previous _tests_? (in which case the logic here renders inappropriate)

Reply:
No, they are root level, and we put them on our behalf.